### PR TITLE
Use --mode [train, test, export] in PT examples, as in TF

### PR DESCRIPTION
**Pull request by vshampor**

### CODE REVIEW

**Comment by vshampor** [comment]
Jenkins please retry a build

File: examples/tensorflow/segmentation/train.py [hunk 1]
```diff
@@ -50,6 +50,15 @@ def get_argument_parser():
                                         cpu_only=False,
                                         metrics_dump=False)
 
+    parser.add_argument(
```
Comment (alexsu52) [inline examples/tensorflow/segmentation/train.py:53]: `train.py` does not require the mode train by design. Please revert this change.

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: examples/tensorflow/segmentation/train.py (hunk 1 above)
Comment (vshampor) [reply examples/tensorflow/segmentation/train.py]: Done

File: examples/tensorflow/segmentation/evaluation.py [hunk 2]
```diff
@@ -46,9 +47,9 @@ def get_argument_parser():
         '--mode',
         '-m',
         nargs='+',
-        choices=['train', 'test', 'export'],
+        choices=['test', 'export'],
```
Comment (alexsu52) [inline examples/tensorflow/segmentation/evaluation.py:50]: What reason for these changes?

File: examples/tensorflow/common/argparser.py [hunk 3]
```diff
@@ -14,12 +14,22 @@
...
 
+    if not without_mode:
+        parser.add_argument(
+            '--mode',
+            '-m',
+            nargs='+',
+            choices=['train', 'test', 'export'],
+            default='train',
+            help='train: performs training and validation; test: tests the model; export: exports the model.'
+        )
```
Comment (alexsu52) [inline examples/tensorflow/common/argparser.py:31]:
```suggestion
     add_argument(
         parser=parser,
         condition=flags.get('mode', True),
         parameters=argument_parameters(
            '--mode',
            '-m',
            nargs='+',
            choices=['train', 'test', 'export'],
            default='train',
            help='train: performs training and validation; test: tests the model; export: exports the model.'))
```

File: examples/tensorflow/segmentation/evaluation.py [hunk 4]
```diff
@@ -35,7 +35,8 @@
 
 
 def get_argument_parser():
-    parser = get_common_argument_parser(weights=False,
+    parser = get_common_argument_parser(without_mode=True,
```
Comment (alexsu52) [inline examples/tensorflow/segmentation/evaluation.py:38]:
```suggestion
    parser = get_common_argument_parser(mode=False,
```

File: examples/tensorflow/segmentation/train.py [hunk 5]
```diff
@@ -42,7 +42,8 @@
 
 
 def get_argument_parser():
-    parser = get_common_argument_parser(weights=False,
+    parser = get_common_argument_parser(without_mode=True,
```
Comment (alexsu52) [inline examples/tensorflow/segmentation/train.py:45]:
```suggestion
    parser = get_common_argument_parser(mode=False,
```

File: examples/tensorflow/common/argparser.py [hunk 6]
```diff
@@ -14,12 +14,22 @@
 from examples.tensorflow.common.sample_config import CustomArgumentParser
 
 
-def get_common_argument_parser(**flags):
+def get_common_argument_parser(without_mode: bool = False, **flags):
```
Comment (alexsu52) [inline examples/tensorflow/common/argparser.py:17]:
```suggestion
def get_common_argument_parser(**flags):
```

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: examples/tensorflow/segmentation/evaluation.py (hunk 2 above)
Comment (vshampor) [reply examples/tensorflow/segmentation/evaluation.py:50]: It's as you've told me, since `train.py` only supports "train" mode, I suppose that `evaluation.py` should only support the other two.

File: examples/tensorflow/segmentation/evaluation.py (hunk 4 above)
Comment (vshampor) [reply examples/tensorflow/segmentation/evaluation.py:38]: Ok, but it really is all the same in Python, especially for the things that most likely will only be changed by us, internally.

File: examples/tensorflow/common/argparser.py (hunk 3 above)
Comment (vshampor) [reply examples/tensorflow/common/argparser.py:31]: Done

File: examples/tensorflow/segmentation/train.py (hunk 5 above)
Comment (vshampor) [reply examples/tensorflow/segmentation/train.py:45]: Done

File: examples/tensorflow/common/argparser.py (hunk 6 above)
Comment (vshampor) [reply examples/tensorflow/common/argparser.py:17]: Done

File: examples/tensorflow/segmentation/evaluation.py (hunk 2 above)
Comment (alexsu52) [reply examples/tensorflow/segmentation/evaluation.py:50]: `evaluation.py` can run in the training mode. In this case, it will listen `train.py` and evaluate checkpoints that were saved by `train.py`.

Comment (vshampor) [reply examples/tensorflow/segmentation/evaluation.py:50]: Made `evaluation.py` use common parser with `mode=True` so that it has all three CLI options for `-m`

File: examples/tensorflow/segmentation/evaluation.py [hunk 7]
```diff
@@ -35,22 +35,14 @@
...
                                         to_h5=False,
                                         dataset_type=False)
 
-    parser.add_argument(
-        '--mode',
-        '-m',
-        nargs='+',
-        choices=['train', 'test', 'export'],
-        default='train',
-        help='train: performs validation during training; test: tests the model; export: exports the model.'
```
Comment (alexsu52) [inline examples/tensorflow/segmentation/evaluation.py:51]: You changed the description string from 'train: performs validation during training; test: tests the model; export: exports the model.' to `train: performs training and validation; test: tests the model; export: exports the model.` which is less informative.

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: examples/tensorflow/segmentation/evaluation.py (hunk 7 above)
Comment (vshampor) [reply examples/tensorflow/segmentation/evaluation.py:55]: Reworded to be more explicit.

**Review by alexsu52 — APPROVED** [review]
